pkg/mock/verifications.go

```diff
@@ -0,0 +1,32 @@
+package mock
+
```

Comment:
And also as discussed IRL, would be nice to have verification of call sequence. I guess maybe it should be here?

Reply:
I'll add this in the next round as fixes as I'm not 100% sure how flexible it can be.